cppcache/integration/test/RegisterKeysTest.cpp

```diff
@@ -39,19 +42,56 @@ namespace {
 using apache::geode::client::binary_semaphore;
 using apache::geode::client::Cache;
 using apache::geode::client::CacheableInt16;
+using apache::geode::client::CacheableInt32;
 using apache::geode::client::CacheableKey;
 using apache::geode::client::CacheableString;
 using apache::geode::client::CacheFactory;
 using apache::geode::client::CacheListenerMock;
+using apache::geode::client::EntryEvent;
 using apache::geode::client::IllegalStateException;
 using apache::geode::client::Region;
+using apache::geode::client::RegionEvent;
 using apache::geode::client::RegionShortcut;
+using apache::geode::client::CacheListener;
 
 using ::testing::_;
 using ::testing::DoAll;
 using ::testing::InvokeWithoutArgs;
 using ::testing::Return;
 
+const int NUMKEYS = 100;
+
+class MyCacheListener : public CacheListener {
```

Comment:
For future reference you should learn and use gmock for this sort of thing. You can mock up the `CacheListener` interface and register the mock with the cache. Then you can assert that specific methods on the mock were invoked.

Reply:
You agreed a real CacheListener with count_down latches was appropriate.